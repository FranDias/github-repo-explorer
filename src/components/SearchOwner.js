import React, { Component } from "react";
import PropTypes from "prop-types";
import debounce from "lodash/debounce";

class SearchOwner extends Component {
  constructor(props) {
    super(props);
    this.onChange = debounce(this.props.onChange, 300);
  }

  static defaultProps = {
    placeholder:
      "Search an organization's repositories, now dispaying HubSpot:",
    onChange: () => {}
  };

  static propTypes = {
    placeholder: PropTypes.string,
    onChange: PropTypes.func.isRequired
  };

  componentWillUnmount() {
    this.onChange.cancel();
  }

  render() {
    const { placeholder } = this.props;

    return (
      <input
        {...{
          type: "text",
          placeholder,
          onChange: e => this.onChange(e.nativeEvent.target.value)
        }}
      />
    );
  }
}

export default SearchOwner;
