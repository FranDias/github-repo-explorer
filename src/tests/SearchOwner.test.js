import React from "react";
import renderer from "react-test-renderer";
import SearchOwner from "../components/Select";

import { mount, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });
describe("Search Owner", () => {
  it("matches its snapshot default props", () => {
    const component = renderer.create(<SearchOwner />);
    expect(component).toMatchSnapshot();
  });

  it("matches its ", () => {
    const component = renderer.create(
      <SearchOwner onChange={() => {}} placeholder="hi" />
    );
    expect(component).toMatchSnapshot();
  });

  it("calls onChange function", () => {
    const testFunction = jest.fn();
    const component = mount(<SearchOwner onChange={testFunction} />);
    component.simulate("focus");
    component.simulate("change", { target: { value: "Changed" } });
    expect(testFunction).toHaveBeenCalled();
  });
  xit("only calls once every 300ms", () => {});
  xit("provides a browser event in callback", () => {});
});
