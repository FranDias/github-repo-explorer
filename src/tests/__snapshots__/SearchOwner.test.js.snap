// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Owner matches its  1`] = `
<select
  name="default"
  onChange={[Function]}
>
  <option
    value="no options provied"
  >
    Sort by 
    no options provied
  </option>
</select>
`;

exports[`Search Owner matches its snapshot default props 1`] = `
<select
  name="default"
  onChange={[Function]}
>
  <option
    value="no options provied"
  >
    Sort by 
    no options provied
  </option>
</select>
`;
